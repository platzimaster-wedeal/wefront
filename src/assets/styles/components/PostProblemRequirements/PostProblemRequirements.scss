@import '../../partials/sizes.scss';
@import '../../partials/colors.scss';
@import '../../Breakpoints.scss';

.post-problem-requirements {

  & strong {
    @include font-size($font-size-normal);
    font-weight: 500;
    color: $color-black-90;
  }

  & ul li {
    margin: $space-size-extra-small 0;
  }

}
