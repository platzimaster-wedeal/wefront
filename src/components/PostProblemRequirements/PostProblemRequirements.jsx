import React from "react";

// Styles
import "../../assets/styles/components/PostProblemRequirements/PostProblemRequirements.scss";
// Components
import Requirements from "../Requirements/Requirements";

const PostProblemRequirements = ({ requirements }) => {
 return (
  <section className="post-problem-requirements">
   <strong className="post-problem-requirements__title">
    Minimun Requirements
   </strong>
   <ul className="post-problem-requirements__list">
    <li>
     <Requirements title={requirements} />
    </li>
   </ul>
  </section>
 );
};

export default PostProblemRequirements;
