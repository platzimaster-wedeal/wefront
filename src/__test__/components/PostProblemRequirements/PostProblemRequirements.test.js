import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import PostProblemRequirements from "../../../components/PostProblemRequirements/PostProblemRequirements";

describe("Test for the component PostProblemRequirements :D", () => {
 const PostProblemRequirementsMounted = mount(<PostProblemRequirements />);
 const PostProblemRequirementsCreated = create(<PostProblemRequirements />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(PostProblemRequirementsCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(PostProblemRequirementsMounted.length).toEqual(1);
  });
  test("Should the title of the component be a string", () => {
   expect(
    typeof PostProblemRequirementsMounted.find(
     ".post-problem-requirements__title"
    ).text()
   ).toEqual("string");
  });
  test("Should the title of the component be Minimun Requirements", () => {
   expect(
    PostProblemRequirementsMounted.find(
     ".post-problem-requirements__title"
    ).text()
   ).toEqual("Minimun Requirements");
  });
 });
});
