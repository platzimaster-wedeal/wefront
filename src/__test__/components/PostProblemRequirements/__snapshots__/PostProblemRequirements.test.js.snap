// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component PostProblemRequirements :D Visual tests for the component should the comopnent have the same tree 1`] = `
<section
  className="post-problem-requirements"
>
  <strong
    className="post-problem-requirements__title"
  >
    Minimun Requirements
  </strong>
  <ul
    className="post-problem-requirements__list"
  >
    <li>
      <div
        className="requirements__container"
      >
        <svg
          className="requirements__container-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <p
          className="requirements__container-content"
        >
          Knowledge about PCs
        </p>
      </div>
    </li>
    <li>
      <div
        className="requirements__container"
      >
        <svg
          className="requirements__container-icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 16.2L4.8 12l-1.4 1.4L9 19 21 7l-1.4-1.4L9 16.2z"
          />
        </svg>
        <p
          className="requirements__container-content"
        >
          Knowledge about PCs
        </p>
      </div>
    </li>
  </ul>
</section>
`;
